api/metadata/models.py

```diff
@@ -0,0 +1,126 @@
+from urllib.parse import urlparse
+
+from core.models import AbstractBaseExportableModel
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+
+from organisations.models import Organisation
+from projects.models import Project
+
+from .fields import GenericObjectID
+
+FIELD_VALUE_MAX_LENGTH = 2000
+
+METADATA_SUPPORTED_MODELS = ["environment"]
+
+# A map of model name to a function that takes the object id and returns the organisation_id
+SUPPORTED_REQUIREMENTS_MAPPING = {
+    "environment": {
+        "organisation": lambda org_id: org_id,
+        "project": lambda project_id: Project.objects.get(
+            id=project_id
+        ).organisation.id,
+    }
+}
+
+
+class FieldType(models.TextChoices):
+    INTEGER = "int"
+    STRING = "str"
```

Comment:
We should split `STRING` into `MULTILINE_STRING` and `SINGLE_LINE_STRING`. Or perhaps just `STRING` and `MULTILINE_STRING` would be sufficient?